### PR TITLE
[velux] Fix limit minimum bug

### PR DESCRIPTION
**Pull request by Chiuaua79**
Fixes issue/bug report #15087

### CODE REVIEW

**Comment by andrewfg** [comment]
@Chiuaua79 can you please share your think, channel and item configuration?

**Review by andrewfg — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxBridgeHandler.java [hunk 1]
```diff
@@ -578,6 +578,9 @@ private void syncChannelsWithProducts() {
                     logger.trace("syncChannelsWithProducts(): channel {} not found.", channelUID);
                     continue;
                 }
+                if (channelUID.getId().equals(VeluxBindingConstants.CHANNEL_ACTUATOR_LIMIT_MINIMUM) || channelUID.getId().equals(VeluxBindingConstants.CHANNEL_ACTUATOR_LIMIT_MAXIMUM)) {
```
Comment (andrewfg) [inline bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxBridgeHandler.java:581]: This is NOT the solution. It will cause those limit channels' state to never be updated in OH.

The outer loop `for (VeluxProduct product : bridgeParameters.actuators.getChannel().existingProducts.valuesOfModified())` iterates over an array of changed channels. And that array should not include the limitXXX channels, because (as you say) those channels should not have had their values modified by the respective API function call.

**Review by andrewfg — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.velux/README.md [hunk 2]
```diff
@@ -238,7 +238,7 @@ The supported Channel and its associated channel type is shown below.
 Unfortunately Velux has decided to closely integrate the rain sensor into the window device.
 The rain sensor is therefore not displayed in the device list.
 On the other hand, the 'limitMinimum' channel of a roof window provides information about rainy weather:
-If it is set internally by the Velux control unit to a value other than zero, it rains. (Joke!!)
+If it is set internally by the Velux control unit to a value other than zero, it rains.
```
[the remark this answers is not in the document]
Comment (andrewfg) [reply bundles/org.openhab.binding.velux/README.md:241]: @Chiuaua79 response..

> Hi Andrew,
>
> English is not my mother tongue, indeed. Dutch t.b.h. I doubt if many other non-native English speakers get the pun. I suggest to rephrase it to prevent a “joke” like it is in the current version, confuses all the users to understand it is a non-factual paragraph and limitMinimum is not related at all to the rain sensors of the window.
>
> Thanks for reviewing. :)
>
> Krgds
> Cor Hoogendoorn

[the remark this answers is not in the document]
Comment (andrewfg) [reply bundles/org.openhab.binding.velux/README.md:241]: I did not write that text; actually @gs4711 wrote it; and I added the 'Joke' suffix because I was not sure what he intended; -- @gs4711 maybe you can clarify it now?

File: bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxBridgeHandler.java (hunk 1 above)
Comment (andrewfg) [reply bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxBridgeHandler.java:581]: @gs4711 do you have any ideas what is going wrong?

**Comment by andrewfg** [comment]
@Chiuaua79 I think I shall need to have a full trace log over some minutes of time in order find out what is happening. Your posts on the forum were helpful but insufficient. The file may be some kilobytes if not megabytes. You can email it to me if you prefer.

**Comment by andrewfg** [comment]
@Chiuaua79 many thanks for the logs.

On reflection I am beginning to hypothesise that 'syncChannelsWithProducts()' should indeed only be synchronizing actual actuator position channels. So indeed, your solution may indeed be (partly) correct; although personally I would suggest to write the code as shown below. However before we commit such a change I would really like to have Guenther's ( @gs4711 ) feedback if he confirms that this hypothesis is indeed correct.

```
In HueBindingConstants.java..

public static final Set<String> synchronizableChannels = Set.of(ACTUATOR_POSITION, ACTUATOR_STATE, ROLLERSHUTTER_POSITION, WINDOW_POSITION, ROLLERSHUTTER_VANE_POSITION, VSHUTTER_POSITION);

In VeluxBridgeHandler.java..

            for (ChannelUID channelUID : BridgeChannels.getAllLinkedChannelUIDs(this)) {
                if (!synchronizableChannels.contains(channelUID.getId())) {
                    continue;
                }
```

**Comment by Chiuaua79** [comment]
Dear Andrew,

I just created the limitMaximum channel and it shows 0 in my case without rain (but as I have the inverted Thing, that would normally be 100 without rain). So the opposite number of the limitMinimum.

In my case, without rain, limitMinimum is 100, with rain 7 (or uninverted 0 and 93). For the “it is raining case” for limitMaximum I currently am not in the position to test. I am leaving for a few days with the Mrs. It has not rained for a month and forecasts in the Netherlands do not predict rain for at least a week :(

Maybe this weekend I can climb the roof to spray the sensor…

On 13 Jun 2023, at 13:45, Andrew Fiddian-Green ***@***.***> wrote:

﻿

@Chiuaua79<https://github.com/Chiuaua79> apropos the wording for the "God makes it rain hypothesis" -- could you please confirm the following? If (when it is not raining) you have limitMinimum=0% and limitMaximum=80% (say) then I suppose that when it does physically rain, limitMinimum changes to 93% and limitMaximum to 100%.

**Comment by Chiuaua79** [comment]
Hi Andrew. Looks good to me. One thing about the readme, I think there is one spelling mistake: “therfore” shouldn’t that spell “therefore”?

On 14 Jun 2023, at 19:35, Andrew Fiddian-Green ***@***.***> wrote:

﻿

@Chiuaua79<https://github.com/Chiuaua79> my proposed changes are here ***@***.***<https://github.com/Chiuaua79/openhab-addons/commit/fbbe360f3b922c9aa76e5ed4bbe21e64351a3c0e>

**Comment by andrewfg** [comment]
The modified files from my PR did not show up here. You probably need to Git Push. Or something like that..

**Comment by Chiuaua79** [comment]
Finally…. Your commits were in my main branch but the pr was linked to a different branch. I had to merge, etc… but in the end there was a run completed without errors :)

On 16 Jun 2023, at 18:26, Andrew Fiddian-Green ***@***.***> wrote:

﻿

The modified files from my PR did not show up here. You probably need to Git Push. Or something like that..

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only two minor README comments.

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by jlaur** [comment]
@Chiuaua79 - can you do a proper signoff with an e-mail address where you can be reached? See https://www.openhab.org/docs/developer/contributing.html#sign-your-work

File: bundles/org.openhab.binding.velux/README.md [hunk 3]
```diff
@@ -235,10 +235,13 @@ The supported Channel and its associated channel type is shown below.
 
 ## Rain Sensor
 
-Unfortunately Velux has decided to closely integrate the rain sensor into the window device.
-The rain sensor is therefore not displayed in the device list.
-On the other hand, the 'limitMinimum' channel of a roof window provides information about rainy weather:
-If it is set internally by the Velux control unit to a value other than zero, it rains. (Joke!!)
+
```
[the remark this answers is not in the document]
Comment (Chiuaua79) [reply bundles/org.openhab.binding.velux/README.md]: Signed-off-by:  Cor Hoogendoorn <chiuaua@hotmail.com>

**Comment by Chiuaua79** [comment]
Dear Jacob,

before I mess things up... I'm only (kind of) familiar with github in the webbrowser. To do a proper sign-off, do I need to "unresolve the conversation"? Would that give me the possibility to do a sign-off again?

Krgds
Cor Hoogendoorn
________________________________

**Comment by jlaur** [comment]
> before I mess things up... I'm only (kind of) familiar with github in the webbrowser. To do a proper sign-off, do I need to "unresolve the conversation"? Would that give me the possibility to do a sign-off again?

I don't think you can fix it through web, at least for this PR, since it involves rewriting Git history. It seems like you need to update your e-mail address in GitHub (sorry, don't know where exactly this is configured) and in your local Git configuration as well, unless you will be manually adding the signoff.

For this PR, if I understand it correctly, @andrewfg actually contributed the final version of the code in commit 1b32745841c7f956206f2e467337f16f42084087? If this is correct, I can use his signoff for the merge commit.

**Comment by Chiuaua79** [comment]
Actually all changes for the Readme and the java are effectively Andrew’s contribution. My only contribution is stirring things up ;)

On 25 Jun 2023, at 15:23, Jacob Laursen ***@***.***> wrote:

﻿

before I mess things up... I'm only (kind of) familiar with github in the webbrowser. To do a proper sign-off, do I need to "unresolve the conversation"? Would that give me the possibility to do a sign-off again?

I don't think you can fix it through web, at least for this PR, since it involves rewriting Git history. It seems like you need to update your e-mail address in GitHub (sorry, don't know where exactly this is configured) and in your local Git configuration as well, unless you will be manually adding the signoff.

For this PR, if I understand it correctly, @andrewfg<https://github.com/andrewfg> actually contributed the final version of the code in commit 1b32745<https://github.com/openhab/openhab-addons/commit/1b32745841c7f956206f2e467337f16f42084087>? If this is correct, I can use his signoff for the merge commit.

**Comment by jlaur** [comment]
@Chiuaua79 - unfortunately this PR's branch now contains unrelated commits.

To fix it, you could have a look here:
https://community.openhab.org/t/rebase-your-code-or-how-to-fix-your-git-history-before-requesting-a-pull/129358

Or perhaps here:
https://github.com/openhab/openhab-addons/pull/13570#issuecomment-1366979136

**Comment by Chiuaua79** [comment]
Well, that went spiffy... Anyway, did a rebase and now everything looks OK, I suppose.

Thank you for bearing with me...

**Comment by andrewfg** [comment]
@Chiuaua79 thank you for your efforts. Probably first steps towards an exciting addiction. Or something..
